New schema with date hassannasserr

hassannasserr opened the pull request.

hassannasserr committed:
close #28 Update models,routes,serializers with the new schema
-make all the required validations
-set the closeDate depend on the status
-set createDate when the complain created

hassannasserr committed:
update getAll function and models and serializers

-make sure that getAll function doesn't return createDate or closeDate

-fix some bugs in the model
-improved  some condintions in serializers

hassannasserr committed:
Merge branch 'main' into new_schema_with_date-hassannasserr

DevAbdoTolba commented on back/app/models.py:
```diff
@@ -19,10 +19,23 @@ def __init__(self, name, age=None, level=None, msg=None, important=False, status
             self.important = important
             self.status = status if status is not None else "open"
+            if createDate:
+                self.createDate = datetime.fromisoformat(createDate)
```
this might return `Wed, 12 May 2004 12:12:12 GMT` which is not an expected return!

DevAbdoTolba commented on back/app/models.py:
```diff
@@ -19,10 +19,23 @@ def __init__(self, name, age=None, level=None, msg=None, important=False, status
+            if status == "open":
+                self.closeDate = None
+            if status == "close" and closeDate:
+                self.closeDate = datetime.fromisoformat(closeDate)
```
this might return Wed, `12 May 2004 12:12:12 GMT` which is not an expected return!

DevAbdoTolba commented on back/app/serializers.py:
```diff
@@ -39,9 +39,26 @@ def serialize_data(data_model):
+         raise SerializationError(" complaints must have create date  .")
+        
+            
+        
```
please make sure that the createDate can not be after the current date :/

DevAbdoTolba reviewed (changes requested):
Great 🫡. small changes to go. keep up 🔥🔥

hassannasserr committed:
Make the required changes and validations in model and serializers

-make the createDate can't be in the future
-change the format of user insertion createDate

hassannasserr committed:
Merge branch 'new_schema_with_date-hassannasserr' of https://github.com/DevAbdoTolba/Pre_AHC-Status_APP into new_schema_with_date-hassannasserr

hassannasserr committed:
Merge branch 'main' into new_schema_with_date-hassannasserr

DevAbdoTolba reviewed (approved):
LGTM 👍
